dask/dataframe/io/parquet.py

```diff
@@ -121,7 +117,8 @@ def read_parquet(path, columns=None, filters=None, categories=None, index=None,
                         columns=[c for c in pf.columns if c in dtypes])
 
     for cat in categories:
```

Comment:
> No, extra columns given to categories are just ignored.

As is, extra columns are ignored by the reader, but are still put in the metadata. This causes problems, as the metadata no longer matches the dtypes/columns of each partition. Example:

```python
In [15]: df = pd.DataFrame({'a': range(5), 'b': range(5)})

In [16]: dd.from_pandas(df, npartitions=2).to_parquet('test.parq')

In [17]: ddf = dd.read_parquet('test.parq', categories=['c'])

In [18]: ddf.dtypes    # Dask thinks there's a column `c`
Out[18]:
a       int64
b       int64
c    category
dtype: object

In [19]: ddf.compute()    # Not actually a column `c`
Out[19]:
       a  b
index
0      0  0
1      1  1
2      2  2
3      3  3
4      4  4
```